e2e/cypress/integration/bot_accounts/crud_spec.js

```diff
@@ -0,0 +1,278 @@
+
+            // * Validate that it's partially updated
+            cy.get('#description').should('have.value', longDescription + '1234');
+
+            // # Update the bot
+            cy.get('#saveBot').click();
+
+            // * Validate that bot saved correctly
+            cy.url().should('include', `/${newTeam.name}/integrations/bots`);
+
+            // * Validate that the description exists
+            cy.findAllByText(longDescription + '1234').should('exist');
+        });
+    });
+
+    it('MM-T1842 Change BOT role', () => {
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // * Check that the previously created bot is listed
+        cy.findByText(`Test Bot (@${botName})`).then((el) => {
+            // # Make sure it's on the screen
+            cy.wrap(el[0].parentElement.parentElement).scrollIntoView();
+
+            // # Click the edit button
+            cy.wrap(el[0].parentElement.parentElement).findByText('Edit').should('be.visible').click();
+
+            // * Validate redirect to edit screen
+            cy.url().
+                should('include', `/${newTeam.name}/integrations/bots/edit`);
+
+            // # Select sysadmin
+            cy.get('select').select('System Admin');
+
+            // * Validate that permissions are set and read only
+            cy.get('#postChannels').should('be.checked').should('be.disabled');
+            cy.get('#postAll').should('be.checked').should('be.disabled');
+
+            // # Update the bot
+            cy.get('#saveBot').click();
+
+            // * Validate that bot saved correctly
+            cy.url().should('include', `/${newTeam.name}/integrations/bots`);
+        });
+    });
+
+    function createBotInteractive(username = `bot-${getRandomId()}`) {
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // # Click add bot
+        cy.get('#addBotAccount').click();
+
+        // # fill+submit form
+        cy.get('#username').type(username);
+        cy.get('#displayName').type('Test Bot');
+        cy.get('#saveBot').click();
+
+        // * verify confirmation page
+        cy.url().
+            should('include', `/${newTeam.name}/integrations/confirm`).
+            should('match', /token=[a-zA-Z0-9]{26}/);
+
+        // * verify confirmation form/token
+        cy.get('div.backstage-form').
+            should('include.text', 'Setup Successful').
+            should((confirmation) => {
+                expect(confirmation.text()).to.match(/Token: [a-zA-Z0-9]{26}/);
+            });
+
+        return cy.get('div.backstage-form').invoke('text');
+    }
+
+    it('MM-T1843 ID along with actual token is created', () => {
+        // # Create the bot and validate token is visible
+        createBotInteractive();
+        cy.get('#doneButton').click();
+    });
+
+    it('MM-T1844 Token is hidden whe you return to the page but ID is still visible', () => {
```

Comment:
```suggestion
    it('MM-T1844 Token is hidden when you return to the page but ID is still visible', () => {
```
Sorry about this Eli! I've corrected this in the test case.